akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -478,30 +469,34 @@ private[remote] abstract class ArteryTransport(_system: ExtendedActorSystem, _pr
     messageDispatcher = new MessageDispatcher(system, provider)
     topLevelFlightRecorder.loFreq(Transport_MaterializerStarted, NoMetaData)
 
-    runInboundStreams()
+    runInboundStreams().map { boundPort =>
+      _bindAddress = UniqueAddress(
```

Comment:
Is that assignment safe? I.e. previously when this method returned `_bindAddress` was already set, now it's only set later from another thread. I wonder if you could make `runInboundStreams` return the port strictly in the first place because we are `Await`ing there anyway?

Reply:
Good question. `bindAddress` is actually only used from `BoundAddressesExtension`, which is actually only used in tests - but technically it's public API. I'll make it an `Option`, that should be safe enough for this use case I suppose.

Reply:
That's probably true.

But wouldn't it be actually simpler (and easier to reason about) to keep the old structure and just return the port strictly from `runInboundStreams`?